src/signalling/peers.rs

```diff
@@ -6,21 +6,58 @@
 use std::{
     collections::HashMap,
     convert::{TryFrom, TryInto},
+    fmt,
 };
 
+use medea_client_api_proto::{Incrementable, PeerId, TrackId};
+
 use crate::{
     api::control::MemberId,
-    media::{Peer, PeerId, PeerStateMachine},
-    signalling::room::RoomError,
+    log::prelude::*,
+    media::{New, Peer, PeerStateMachine},
+    signalling::{elements::Member, room::RoomError},
 };
 
 #[derive(Debug)]
 pub struct PeerRepository {
     /// [`Peer`]s of [`Member`]s in this [`Room`].
     ///
-    /// [`Member`]: crate::api::control::member::Member
+    /// [`Member`]: crate::signalling::elements::member::Member
     /// [`Room`]: crate::signalling::Room
     peers: HashMap<PeerId, PeerStateMachine>,
+
+    /// Count of [`Peer`]s in this [`Room`].
+    ///
+    /// [`Room`]: crate::signalling::room::Room
+    peers_count: Counter<PeerId>,
+
+    /// Count of [`MediaTrack`]s in this [`Room`].
+    ///
+    /// [`MediaTrack`]: crate::media::track::MediaTrack
+    /// [`Room`]: crate::signalling::room::Room
+    tracks_count: Counter<TrackId>,
+}
+
+/// Simple ID counter.
+#[derive(Default, Debug)]
+pub struct Counter<T: Incrementable + Copy> {
```

Comment:
This bound is redudant. Consider [bounding behavior, not data](https://github.com/instrumentisto/rust-incubator/tree/92d5e403360bcad364232aada732ad70f4101011/2_idioms/2_3_bound_impl).